app/scripts/models/mixins/validate.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin to validate data. Uses the [VAT
+ * library](https://github.com/shane-tomlinson/vat) to
+ * validate/transform data.
+ *
+ * If a required field is missing from the data, a
+ * `MISSING_ERROR` error is generated, with the error's
+ * `param` field set to the missing field's name.
+ *
+ * If a field does not pass validation, an
+ * `INVALID_PARAMETER` error is generated, with the error's
+ * `param` field set to the invalid field's name.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var Vat = require('vat');
+
+  module.exports = {
+    /**
+     * Validate data against a schema
+     *
+     * @param {object} data - data to validate
+     * @param {object} schema - schema that can be passed to the validator
+     * @param {object} Errors - Errors module used to create errors
+     * @returns {object} validation/transformation results
+     */
+    validateUsingSchema: function (data, schema, Errors) {
```

Comment:
In light of the transformation stuff, I wonder if it's possible to come up with a slightly more descriptive name for this, one that indicates data in !== data out?

I haven't come up with anything myself yet, but I'm having a think...